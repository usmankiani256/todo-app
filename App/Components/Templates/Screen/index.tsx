import { ScrollView, View, SafeAreaView } from 'react-native'
import useStyles from './styles'
import React from 'react'

type ScrollProps = React.ComponentProps<typeof ScrollView>

interface ScreenProps extends ScrollProps {
  children: React.ReactNode
}

const Screen = (props: ScreenProps) => {
  const { children } = props

  const { root, rootScroll } = useStyles()

  return (
    <ScrollView
      {...props}
      keyboardShouldPersistTaps={'handled'}
      contentInsetAdjustmentBehavior="automatic"
      style={rootScroll}>
      <SafeAreaView>
        <View style={root}>{children}</View>
      </SafeAreaView>
    </ScrollView>
  )
}

export default Screen
