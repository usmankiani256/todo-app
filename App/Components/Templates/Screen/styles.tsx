import { Dimensions, StyleSheet } from 'react-native'
import { Colors } from 'Theme'

const { width, height } = Dimensions.get('window')

const useStyles = () =>
  StyleSheet.create({
    rootScroll: {
      flexGrow: 1,
      backgroundColor: Colors.foreground,
    },
    root: {
      flex: 1,
    },
  })

export default useStyles
