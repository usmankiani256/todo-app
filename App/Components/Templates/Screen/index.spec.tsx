import * as React from 'react'
import { render } from '@testing-library/react-native'
import { Text } from '@Atoms'
import Screen from './index'

describe('Screen Component', () => {
  it('renders correctly', () => {
    const rendered: any = render(
      <Screen>
        <Text>Hello, world!</Text>
      </Screen>,
    ).toJSON()

    expect(rendered).toMatchSnapshot()
  })

  it('renders props', () => {
    const rendered: any = render(
      <Screen scrollEnabled={false}>
        <Text>Hello, world!</Text>
      </Screen>,
    ).toJSON()

    expect(rendered).toMatchSnapshot()
  })
})
