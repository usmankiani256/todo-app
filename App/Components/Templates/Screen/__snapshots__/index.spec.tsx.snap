// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen Component renders correctly 1`] = `
<RCTScrollView
  contentInsetAdjustmentBehavior="automatic"
  keyboardShouldPersistTaps="handled"
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flexGrow": 1,
    }
  }
>
  <View>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#3f3f3f",
              "fontFamily": "SofiaPro",
              "fontSize": 20,
              "lineHeight": 20,
            }
          }
        >
          Hello, world!
        </Text>
      </View>
    </RCTSafeAreaView>
  </View>
</RCTScrollView>
`;

exports[`Screen Component renders props 1`] = `
<RCTScrollView
  contentInsetAdjustmentBehavior="automatic"
  keyboardShouldPersistTaps="handled"
  scrollEnabled={false}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flexGrow": 1,
    }
  }
>
  <View>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#3f3f3f",
              "fontFamily": "SofiaPro",
              "fontSize": 20,
              "lineHeight": 20,
            }
          }
        >
          Hello, world!
        </Text>
      </View>
    </RCTSafeAreaView>
  </View>
</RCTScrollView>
`;
